public/media_engagement/imaqal_sms_traffic.js

```diff
@@ -769,20 +769,23 @@ const update = (data) => {
     var fullDateFormat = d3.timeFormat("%c")	
 
     // Update timestamp of update and reset formatting
-    const timestamp = new Date()
-    d3.select("#lastUpdated").classed("alert", false).text(fullDateFormat(timestamp))
+    const timestamp = new Date(Math.max.apply(null, data.map(function(d) {
+        return new Date(d.datetime);
+    })));
+    d3.select("#lastUpdated").classed("text-danger", false).text(fullDateFormat(timestamp))
 
     function setLastUpdatedAlert() {
         // Calculate time diff bw current and timestamp
         var currentTime = new Date()
         var difference_ms = (currentTime.getTime() - timestamp.getTime())/60000
         var difference_minutes = Math.floor(difference_ms % 60)
         // if updated more than 20 min ago >> reformat
-        if (difference_minutes > 20) {
-            d3.select("#lastUpdated").classed("alert", true)
+        if (difference_minutes > 30) {
+            d3.select("#lastUpdated").classed("text-danger alert alert-danger", true)
         }
     };
 
     setInterval(setLastUpdatedAlert, 1000)
 };
 
+
```

Comment:
please remove unnecessary new line

Reply:
This is still pending @DanielMwendwa

Reply:
I've checked the code, I'd already removed the new line